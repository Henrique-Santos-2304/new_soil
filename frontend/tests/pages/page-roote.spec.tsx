import Home from '@/pages/index'
import '@testing-library/jest-dom'
import { renderThemeMock } from '../mocks'

describe('Home', () => {
  it('renders a this component to be defined', () => {
    const { container } = renderThemeMock(<Home />)

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should expect to have header component ', () => {
    const { container } = renderThemeMock(<Home />)

    const header = container.querySelector('header[id="header-of-application')

    expect(header).toBeInTheDocument()
  })
})
