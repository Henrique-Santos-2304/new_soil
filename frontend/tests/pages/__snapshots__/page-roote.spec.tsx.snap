// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a this component to be defined 1`] = `
<div>
  <div
    class="MuiBox-root css-imz5j4"
  >
    <header
      aria-label="Cabeçalho da aplicação"
      class="MuiBox-root css-0"
      id="header-of-application"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 MuiBox-root css-16bjjll-MuiPaper-root"
      >
        <button
          aria-label="botão para fazer logout na aplicação"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1xri6ks-MuiButtonBase-root-MuiButton-root"
          id="btn-logout"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Sair
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root css-49rcs4"
          id="titulo e subtitulo da página"
        >
          <h1
            aria-label="Título da página"
            class="MuiTypography-root MuiTypography-h1 css-7inatm-MuiTypography-root"
          >
            Pivô 1
          </h1>
          <div
            aria-label="Sub-Título da página"
            class="MuiBox-root css-ny6if"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-10sbftd-MuiTypography-root"
            >
              Ultima Atualização: 
            </h4>
            <h4
              class="MuiTypography-root MuiTypography-h4 css-tb5xb8-MuiTypography-root"
            >
              05/01/2023 09:02:40
            </h4>
          </div>
        </div>
        <button
          class="MuiBox-root css-13q49m9"
          id="btn-theme-container"
        >
          <svg
            aria-hidden="true"
            arial-label="botão que muda o tema global do site para claro"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-pnw5j7-MuiSvgIcon-root"
            data-testid="LightModeIcon"
            focusable="false"
            id="icon-light"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41l-1.06-1.06zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
            />
          </svg>
          <h6
            aria-label="Texto indicando que o botão acima é usado para mudar o tema"
            class="MuiTypography-root MuiTypography-h6 css-1khoe0w-MuiTypography-root"
            id="txt-change-theme"
          >
            Mudar Tema
          </h6>
        </button>
      </div>
    </header>
  </div>
</div>
`;
